src/procedures/utility/build_weighted_matrix.c

```diff
@@ -0,0 +1,475 @@
+			Graph_GetEdge(contx->g, CurrID, &currE);                           \
+			currV = AttributeSet_Get(*currE.attributes, contx->w);             \
+			if(currV != ATTRIBUTE_NOTFOUND && (SI_NUMERIC & SI_TYPE(*currV))   \
+				&& BWM_COMP(minV, *currV)                                      \
+			)                                                                  \
+			{                                                                  \
+				minV = *currV;                                                 \
+				minID = CurrID;                                                \
+			}                                                                  \
+			info = GxB_Vector_Iterator_next(i);                                \
+		}                                                                      \
+		*z = (uint64_t) minID;                                                 \
+		GxB_Iterator_free(&i);                                                 \
+	}                                                                          \
+}
+typedef struct
+{
+	const Graph *g;		// Graph
+	AttributeID w;	// Weight Edge
+	RelationID rel;
+} compareContext;
+
+void _selectAttribute
+(
+	bool *z,
+	const uint64_t *x,
+	GrB_Index i,
+	GrB_Index j,
+	const 	compareContext *contx
+)
+{
+	Edge _x;
+	Graph_GetEdge(contx->g, (EdgeID) (*x), &_x);
+	SIValue *v = GraphEntity_GetProperty((GraphEntity *) &_x, contx->w);
+	*z = (v != ATTRIBUTE_NOTFOUND) && (SI_NUMERIC & SI_TYPE(*v));
+}
+void _getAttFromID
+(
+	double *z, 
+	const uint64_t *x, 
+	const compareContext *contx
+)
+{
+	Edge _x, _y;
+	Graph_GetEdge(contx->g, (EdgeID) (*x), &_x);
+	SIValue *v = AttributeSet_Get(*_x.attributes, contx->w);
+	ASSERT(v != ATTRIBUTE_NOTFOUND);
+	int info = SIValue_ToDouble(v, z);
+	ASSERT(info == 1);
+}
+
+#define BWM_COMP(X,Y) SIValue_Compare((X), (Y), NULL) == 1
+#define BWM_INF INFINITY
+void _minGetEdgeID 
+COMPAREEDGES
+
+void _reduceToMatrixMin
+TENSORPICK
+
+#undef BWM_COMP
+#undef BWM_INF
+#define BWM_COMP(X,Y) SIValue_Compare((X), (Y), NULL) == -1
+#define BWM_INF -INFINITY
+void _maxGetEdgeID 
+COMPAREEDGES
+
+void _reduceToMatrixMax
+TENSORPICK
+
+void _reduceToMatrixAny
+(
+	uint64_t *z, 
+	const uint64_t *x
+)
+{
+	if(SCALAR_ENTRY(*x))
+	{
+		*z = *x;
+	}
+	else
+	{
+		GrB_Vector _v = AS_VECTOR(*x);
+		GxB_Iterator i = NULL;
+		GrB_Info info = GxB_Iterator_new(&i);
+		ASSERT(info == GrB_SUCCESS);
+		info = GxB_Vector_Iterator_attach(i, _v, NULL);
+		ASSERT(info == GrB_SUCCESS);
+		info = GxB_Vector_Iterator_seek(i, 0);
+		ASSERT(info == GrB_SUCCESS);
+		EdgeID minID = (EdgeID) GxB_Vector_Iterator_getIndex(i);
+		info = GxB_Iterator_free(&i);
+		ASSERT(info == GrB_SUCCESS);
+		*z = minID;
+	}
+}
+// compose multiple label & relation matrices into a single matrix
+// L = L0 U L1 U ... Lm
+// A = L * (R0 + R1 + ... Rn) * L
+//
+// rows = L's main diagonal
+// in case no labels are specified rows is a dense 1 vector: [1,1,...1]
+GrB_Info Build_Weighted_Matrix
```

Comment:
comment should mention where do we keep the weights and how do we resolve multiple weights mapped to the same entry.